### ISSUE TITLE
Warden Anti-cheat client crashes

### ISSUE THREAD

**Issue by Exanimous** [CoronaCore/Issues#3]
After some implementation and testing of the warden implementation on CMANGOS, I have concluded that compiling in release mode will cause a client crash on login attempt if warden is activated.

I think I have narrowed the issue to void WorldSession::SendPacket(WorldPacket const\* packet)

if release mode is active, then the following code is never called:

if (m_Socket->SendPacket(*packet) == -1)
        m_Socket->CloseSocket();

The first request from warden that causes this client crash seems to be from -
void WardenBase::RequestModule(), and judging by the server logs this happens immediatly on log in attempt.

Is it possible that this bug is caused by warden polling a check too early before a client is proplerly logged into the world, causing a socket to be left open (in release mode).

The client crash has the following info:

Improper header received: [ CE FA ED FE 07 00 00 00 03 00 00 00 08 00 00 00 0A 00 00 00 48 07 00 00 85 20 01 00 01 00 00 00 

And the threads seem to get stuck in : 

KERNELBASE.dll WaitForSingleObject+18

**Comment by Salja** [comment]
Tested and no crashes in release mode

**Comment by Salja** [comment]
Is Fixed you tested ?

**Comment by Exanimous** [comment]
After changing my work enviroment (win8 to wn7) and disabling anti-virus protection, I have not been able to replicate the issue.  I have tested it again on win8 64bit with VS2010 and I have not had a repeat of the same problem.

It is however very strange that the small change in the code above prevented client crash. (A change I have now reverted)
I will keep an eye on it and report again if I have any more issues, but for now it seems to work well.
